# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.h
# Commit: a69c40accbf0
# Full Hash: a69c40accbf099d041417ce4da9b82829f044167
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2023-12-06 21:30:12
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a bug where caret misses when dragging and dropping r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192377
# ==============================================================================

diff -r c9558e7c5c2d -r a69c40accbf0 layout/painting/nsDisplayList.h
--- a/layout/painting/nsDisplayList.h	Wed Dec 06 16:17:57 2023 +0000
+++ b/layout/painting/nsDisplayList.h	Wed Dec 06 16:29:18 2023 +0000
@@ -656,7 +656,7 @@
    * Get the frame that the caret is supposed to draw in.
    * If the caret is currently invisible, this will be null.
    */
-  nsIFrame* GetCaretFrame() { return mCaretFrame; }
+  nsIFrame* GetCaretFrame() { return CurrentPresShellState()->mCaretFrame; }
   /**
    * Get the rectangle we're supposed to draw the caret into.
    */
@@ -1726,6 +1726,7 @@
     bool mInsidePointerEventsNoneDoc;
     bool mTouchEventPrefEnabledDoc;
     nsIFrame* mPresShellIgnoreScrollFrame;
+    nsIFrame* mCaretFrame = nullptr;
   };
 
   PresShellState* CurrentPresShellState() {
@@ -1760,7 +1761,6 @@
   // The reference frame for mCurrentFrame.
   const nsIFrame* mCurrentReferenceFrame;
 
-  nsIFrame* mCaretFrame;
   // A temporary list that we append scroll info items to while building
   // display items for the contents of frames with SVG effects.
   // Only non-null when ShouldBuildScrollInfoItemsForHoisting() is true.
